### PR TITLE
512 lab 7 redux conversion

### LINKED ISSUE #512: Lab 7 Redux conversion
# Lab 1 Redux conversion

As a **developer**, I want **to use native react context and reducers**, so that **they are consistent for all labs**.

Associated Epic: [#496](https://github.com/all-rit/ALL/issues/496)

## Definition of Done

- [ ] Acceptance criteria defined
- [ ] Solution tasks specified
- [ ] Feature branch created
- [ ] ~~Unit tests written (on API code only for now)~~
- [ ] ~~Solution passes full suite of unit tests~~
- [ ] Design documents updated if needed
- [ ] Pull request created
- [ ] Code is reviewed
- [ ] Code is merged to `develop` branch
- [ ] ~~Tests pass on CI~~
- [ ] ~~Deployed to staging environment~~
- [ ] Acceptance criteria is met

## Acceptance Criteria

- [x] Given **I am in the exercise**, when **I try to exit the activity**, then **then I am forced to stay in the exercise**
- [x] Given **when I am in the**, when **I change sections**, then **I am given the correct playing state information**

## Solution Tasks

- [ ] [Task 1]
- [ ] [Task 2]
- [ ] [Task 3]

### PR DESCRIPTION
**Pull request by Mster115**
**NOTE: I did not convert the repair to the new model for lab 7. I was able to easily convert the repair for this one**

- [ ] No discrepancies across browsers. Eg. chrome vs safari
- [ ] Pages pass W3C Validation for HTML (https://validator.w3.org)
- [ ] CSS (https://jigsaw.w3.org/css-validator/)
- [ ] Google sign-in is operational
- [ ] Accessibility functions
- [ ] Pages can scale without distorting page
- [ ] No dead links
- [ ] Navbar is consistent across the site
- [ ] Pages are screen-reader accessible
- [ ] Contrast meets standards for accessibility
- [ ] Pages are keyboard accessible
- [ ] Passes WAVE Evaluation Tool chrome extension

### CODE REVIEW

File: .vscode/settings.json
Comment (heathermoses) [inline .vscode/settings.json:1]: nit: shouldn't be in PR

**Review by heathermoses — APPROVED** [review]
I went through the entire lab and found that everything works as expected!

**Review by AndreasLc1103 — APPROVED** [review]
Reviewed and Looks good to me. Also ran it to make sure things are good overall

**Review by heathermoses — APPROVED** [review]
LGTM

**Pull request merged**